sdk/resourcemanager/Azure.ResourceManager/docs/MigrationGuide.md

```diff
@@ -0,0 +1,287 @@
+using Azure.Identity;
+using Azure.ResourceManager.Core;
+using Azure.ResourceManager.Compute;
+using Azure.ResourceManager.Network;
+using System;
+using System.Threading.Tasks;
+```
+
+### Setting up the clients
+#### Old
+```C#
+ServiceClientCredentials credentials = getMyCredentials();
+var computeClient = new ComputeManagementClient(credentials);
+var networkClient = new NetworkManagementClient(credentials);
+var managedServiceIdentityClient = new ManagedServiceIdentityClient(credentials);
+```
+#### New
+```C#
+var armClient = new ArmClient(new DefaultAzureCredential());
+```
+As you can see, authentication is now handled by Azure.Identity, and now just a single client is needed, from which you can get the `DefaultSubscription` and start managing your resources. 
+
+### Create a Resource Group
+#### Old
+```C#
+ServiceClientCredentials credentials = getMyCredentials(); 
+var resourcesClient = new ResourceManagementClient(credentials);
+
+string rgName = "QuickStartRG";
+string location = "WestUS2";
+resourcesClient.ResourceGroups.CreateOrUpdate(
+    rgName,
+    new ResourceGroup
+    {
+        Location = location,
+        Tags = new Dictionary<string, string>() { { rgName, DateTime.UtcNow.ToString("u") } }
+    });
+```
+#### New
+```C#
+var armClient = new ArmClient(new DefaultAzureCredential());
+Subscription subscription = armClient.DefaultSubscription;
+ResourceGroupContainer rgContainer = subscription.GetResourceGroups();
+
+Location location = Location.WestUS2;
+string rgName = "QuickStartRG";
+ResourceGroup resourceGroup = await rgContainer.Construct(location).CreateOrUpdateAsync(rgName)
+```
+
+In the new libraries, the Resource Group is created trought a container that came from the subscription. Also, public locations are provided trought the `Location` object, but it can be specified as a string as well. Additionally, there's no need to create a `ResourceGroup` object to send parameters when creating a resource group, since tags can be added directly from the resource group variable and the location is specified in the `Construct()` method.
+
+### Create an Availability Set
+#### Old
+```C#
+var inputAvailabilitySet = new AvailabilitySet
+{
+    Location = location,
+    Tags = new Dictionary<string, string>()
+    {
+        { "RG", "rg" },
+        { "testTag", "1" }
+    },
+    PlatformFaultDomainCount = 1,
+    PlatformUpdateDomainCount = 1,
+    Sku = new CM.Sku
+    {
+        Name = AvailabilitySetSkuTypes.Aligned 
+    }
+};
+string aSetName = "quickstartvm_aSet";
+
+var asCreateOrUpdateResponse = computeClient.AvailabilitySets.CreateOrUpdate(
+    rgName,
+    aSetName,
+    inputAvailabilitySet
+);
+string aSetID = $"/subscriptions/{computeClient.SubscriptionId}/resourceGroups/{rgName}/providers/Microsoft.Compute/availabilitySets/{aSetName}";
+```
+#### New
+```C#
+string vmName = "quickstartvm";
+var aset = await resourceGroup.GetAvailabilitySets().Construct("Aligned").CreateOrUpdateAsync(vmName + "_aSet");
```

Comment:
same comment on construct, we probably need to change this in the [samples ](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/resourcemanager/Azure.ResourceManager/samples/Sample3_CreatingAVirtualNetwork.md)too.